lib/yt_search/subtitles.ex

```diff
@@ -28,26 +27,56 @@ defmodule YtSearch.Subtitle do
     SubtitleRepo.replica().all(query)
   end
 
+  def blob(nil), do: nil
+
+  def blob(%__MODULE__{} = subtitle) do
+    blob(subtitle_id(subtitle))
+  end
+
+  def blob(id) when is_bitstring(id) do
+    case File.read(path_for(id)) do
+      {:ok, data} -> data
+      {:error, :enoent} -> nil
+    end
+  end
+
+  def path_for(id) do
+    "subtitles/#{id}"
+  end
+
+  def subtitle_id(%__MODULE__{youtube_id: youtube_id, language: language}) do
+    "#{youtube_id}_#{language}"
+  end
+
+  def subtitle_data(%__MODULE__{} = subtitle) do
+    blob(subtitle)
+  end
+
   @spec insert(String.t(), String.t(), String.t() | nil, map()) :: t()
   def insert(youtube_id, language, subtitle_data, cta) do
-    %__MODULE__{
+    subtitle = %__MODULE__{
       youtube_id: youtube_id,
       language: language,
-      subtitle_data: subtitle_data,
       cta: cta
     }
     |> SubtitleRepo.insert!(
       on_conflict: [
         set: [
-          subtitle_data: subtitle_data,
           cta: cta
         ]
       ]
     )
+
+    if subtitle_data do
+      File.mkdir_p!("subtitles")
```

Comment:
nitpick optimization: `File.mkdir_p!` can be called at the `YtSearch.Application` module, since it only really needs to run once on app startup, thumbnail folder is initialized this way as well